### PR TITLE
Permissions react

### PR DESCRIPTION
**Pull request by garrensmith**
Permissions for a database done in React

### CODE REVIEW

**Comment by benkeen** [comment]
Code looks great, but I'll look it over a little longer. One minor UX niggle: after adding an item in any of the four sections on the Permissions page, it should probably remove the text you just entered and re-focus on the field.

**Comment by michellephung** [comment]
+1

**Comment by benkeen** [comment]
Besides previous comment, +1

File: app/addons/permissions/components.react.jsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        type: 'roles',
+        section: this.props.section,
+        value: this.state.newRole
+      });
+
+      this.setState({newRole: ''});
+    },
+
+    getItems: function (items, type) {
+      return _.map(items, function (item, i) {
```
Comment (robertkowalski) [inline app/addons/permissions/components.react.jsx]: you can use native map, i think we can remove underscore/lodash in the future as those features are all now implemented into browsers.

http://kangax.github.io/compat-table/es5/#Array.prototype.map

**Comment by michellephung** [comment]
if your going to change the UI, I'd reword the sub-line under "Roles" to:
 `All users with any of the following role(s) will have admins access.`,
or
`Users with the following role(s) will have admins access.`

it's unclear if you need all of the roles to have admin access, or just one of the roles.

**Comment by michellephung** [comment]
i think it's `any of the following`

**Comment by benkeen** [comment]
Sorry to be a pain, one more minor thing: it allows you to add a blank entry. I think this is an old bug. The UI looks a little wonky due to the empty user name. 

screen shot 2015-05-19 at 9 24 49 am

**Comment by garrensmith** [comment]
Issued fixed and merged.
